### PR TITLE
support -D parameter for auth&runtime

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java [hunk 1]
```diff
@@ -339,6 +340,7 @@ protected Azure getOrCreateAzureClient() throws AzureAuthFailureException, Azure
         try {
             final MavenAuthConfiguration mavenAuthConfiguration = auth == null ? new MavenAuthConfiguration() : auth;
             mavenAuthConfiguration.setType(getAuthType());
```
Comment (wangmingliang-ms) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java:342]: this should be done immediately after the mojo properties are ready.

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/utils/SystemPropertyUtils.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import java.lang.reflect.Modifier;
+import java.util.List;
+
+public class SystemPropertyUtils {
+    public static void injectCommandLineParameter(String prefix, Object obj, @Nonnull Class cls) {
+        Object result = obj;
+        try {
+            if (result == null) {
+                try {
+                    result = cls.getDeclaredConstructor().newInstance();
```
Comment (wangmingliang-ms) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/utils/SystemPropertyUtils.java:25]: result is never used.

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java [hunk 3]
```diff
@@ -339,6 +340,7 @@ protected Azure getOrCreateAzureClient() throws AzureAuthFailureException, Azure
...
+            SystemPropertyUtils.injectCommandLineParameter("auth", mavenAuthConfiguration, MavenAuthConfiguration.class);
```
Comment (wangmingliang-ms) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java:343]: will `mavenAuthConfiguration` be null?
what happens if `mavenAuthConfiguration` is null, it seems your implementation does not support null.
and if yes, the third parameter is not needed.

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/utils/SystemPropertyUtils.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import org.apache.commons.lang3.reflect.FieldUtils;
+
+import javax.annotation.Nonnull;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Modifier;
+import java.util.List;
+
+public class SystemPropertyUtils {
+    public static void injectCommandLineParameter(String prefix, Object obj, @Nonnull Class cls) {
```
Comment (wangmingliang-ms) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/utils/SystemPropertyUtils.java:20]: if `obj` is null, all the tasks done is this method is useless and affect nothing.

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java (hunk 3 above)
Comment (andxu) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java:343]: mavenAuthConfiguration should be always non-null, but the other parameter runtime may be null

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**